modules/my_health/spec/request/v1/messaging_preferences_spec.rb

```diff
@@ -8,100 +8,148 @@
+      end
 
-    it 'responds to GET #show of preferences' do
-      VCR.use_cassette('sm_client/preferences/fetches_email_settings_for_notifications') do
-        get '/my_health/v1/messaging/preferences'
+      it 'requires all parameters for update' do
+        VCR.use_cassette('sm_client/preferences/sets_the_email_notification_settings', record: :none) do
+          params = { email_address: 'kamyar.karshenas@va.gov' }
+          put '/my_health/v1/messaging/preferences', params:
+        end
+
+        expect(response).to have_http_status(:unprocessable_entity)
       end
 
-      expect(response).to be_successful
-      expect(response.body).to be_a(String)
-      attrs = JSON.parse(response.body)['data']['attributes']
-      expect(attrs['email_address']).to eq('muazzam.khan@va.gov')
-      expect(attrs['frequency']).to eq('daily')
-    end
+      it 'rejects unknown frequency parameters' do
+        VCR.use_cassette('sm_client/preferences/sets_the_email_notification_settings', record: :none) do
+          params = { email_address: 'kamyar.karshenas@va.gov',
+                     frequency: 'hourly' }
+          put '/my_health/v1/messaging/preferences', params:
+        end
 
-    it 'responds to PUT #update of preferences' do
-      VCR.use_cassette('sm_client/preferences/sets_the_email_notification_settings', record: :none) do
-        params = { email_address: 'kamyar.karshenas@va.gov',
-                   frequency: 'none' }
-        put '/my_health/v1/messaging/preferences', params:
+        expect(response).to have_http_status(:unprocessable_entity)
       end
 
-      expect(response).to have_http_status(:ok)
-      expect(JSON.parse(response.body)['data']['id'])
-        .to eq('17126b0821ad0472ae11944e9861f82d6bdd17801433e200e6a760148a4866c3')
-      expect(JSON.parse(response.body)['data']['attributes'])
-        .to eq('email_address' => 'kamyar.karshenas@va.gov', 'frequency' => 'none')
-    end
+      it 'returns a custom exception mapped from i18n when email contains spaces' do
+        VCR.use_cassette('sm_client/preferences/raises_a_backend_service_exception_when_email_includes_spaces') do
+          params = { email_address: 'kamyar karshenas@va.gov',
+                     frequency: 'daily' }
+          put '/my_health/v1/messaging/preferences', params:
+        end
 
-    it 'requires all parameters for update' do
-      VCR.use_cassette('sm_client/preferences/sets_the_email_notification_settings', record: :none) do
-        params = { email_address: 'kamyar.karshenas@va.gov' }
-        put '/my_health/v1/messaging/preferences', params:
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(JSON.parse(response.body)['errors'].first['code']).to eq('SM152')
       end
+    end
+  end
 
-      expect(response).to have_http_status(:unprocessable_entity)
+  context 'when legacy sm policy' do
+    before do
+      Flipper.disable(:mhv_sm_session_policy)
```

Comment:
same here, is you allow(Flipper)... you won't have the possibility of running into flaky specs